### DISCUSSION TITLE
Questions on System Requirements

### DISCUSSION THREAD

**Discussion by matt-ashman** [Sky360-Repository/sky360#5] [Q&A]
Hi Folks,

I'm a noob and trying to catch up as quickly as possible.

I read the guide below and didn't find the answers to a couple of my questions:
[Sky360_Design_Developement_Guide.pdf](https://github.com/Sky360-Repository/sky360/files/9836004/Sky360_Design_Developement_Guide.pdf)

Can you please help me learn and create a sort of document for others like me?

Maybe we could populate the wiki with whatever is written below?

Two questions I have to get it started:

- I've heard that 15km between sensors is the best distance. Is there documentation of how that was determined?
- What kind of time synchronization is required between sensors for expected UAP motions?

Thanks!

**Comment by i-am-logger** [comment]
I can't answer the first question. I believe @RichardGHopf can put some insights on it. I believe it to be done with SkyHub, prior to Sky360. 

As for the second question. The goal is to use GPS signal(from the GPS unit) as a Time Source for ROS2 to achieve real time and synchronization/ordering. https://design.ros2.org/articles/clock_and_time.html

**Reply by matt-ashman** [reply]
Nice, thank you for sharing that.

Do we already have a sense of how accurate that solution could be in the best case?

The paper on Ukrainian UAP observations speaks to timing requirements:

"The exposure time was chosen so that the image of the object did not shift significantly during exposure. The frame rate was chosen to take into account the speed of the object and the field of view of the camera. In practice, the exposure time was less than 1 ms, and the frame rate was no less than 50 Hz. Frames were recorded in the .ser format with 14 and 16 bits. Violation of these conditions leads to the fact that objects will not be registered during observations. To determine the coordinates of objects, the cameras were installed in the direction of the zenith or the Moon."

https://arxiv.org/pdf/2208.11215.pdf

Assuming 1 ms exposure time (for example) could it be we need even more accurate sync for good triangulation?

**Reply by i-am-logger** [reply]
Yes, the solution , with time, will be real-time . 

For now we are defining the robot parts and focus on the integration of the all sky camera, UI, tracker to a somewhat workable version . 

Since all is based on ros2 and ros2 humble is based on fastDDS. We have the real-time requirements embedded as part of the protocol. So it will be further optimization . 

In order to answer the timing problem the system must be a real-time one. When you have to respect timing and ensuring reliability .

**Reply by matt-ashman** [reply]
Makes sense. What about the relative time sync between stations?

Do we have a sense of how good that would be in the best case with ROS-based GPS time and DDS?

**Reply by i-am-logger** [reply]
That will depend mostly on the connectivity . Though can still achieve real-time sub second over internet.

**Reply by matt-ashman** [reply]
Imagine that we are one day processing events/tracks from individual stations with shared compute, for example using a cloud-backed message queue and worker(s) ingesting and updating 3D tracks based on this data.

I am wondering how relatively accurate our detection timestamps would be for the purposes of triangulation between stations? Do we have a way to estimate that?

**Reply by matt-ashman** [reply]
Oh check this out, were you already aware of this? Haven't seen this before.
https://chrony.tuxfamily.org/

Found it here:
https://answers.ros.org/question/11570/ros-time-across-machines/

And according to https://en.wikipedia.org/wiki/Network_Time_Protocol:
"NTP can usually maintain time to within tens of milliseconds over the public internet, and can achieve better than one millisecond accuracy in local area networks under ideal conditions."

Actually 1 ms sounds promising, but that's on LAN "under ideal conditions" which I assume means no collisions.

**Reply by i-am-logger** [reply]
Yea,

The idea is GPS -> NTP -> ROS Time Source

**Comment by Sky360Admin** [comment]
15km? you heard from our side, or from somewhere else?
we go for the 30km mean distance between sensors as a compromise in number of stations needed and accuracy to achieve. as we use several sensors with different angular resolution and range, there's a sweetspot at 30km, that makes sense photogrammetry-wise as atmospheric-wise. when you put a sensor at eg. 2m above ground (which is just a good guess) the horizon is between 30-40km in distance (if there are no mountains, hills and woods).
if you take into account, that the atmosphere is anisotrop and has a lensing effect, you know, that events between horizon and 30° elevation angle are considered as impossible to get high accuracy in triangulation.
the atmosphere becomes thinner when you go further to the zenith, where we can easily see the ISS on 422km distance. but hoizontally is a bit of problem optically. that's why some people are working on passive radar for Sky360 (lead Prof. Bob McGwier, USA) to overcome the optical boundaries.
back to your question ... 15km not from us ... 30km is our chosen, good-compromise distance.
Richard.

**Reply by matt-ashman** [reply]
Thank you for the detailed answer Richard. I have a few follow up questions.

Have we determined the minimum useful distance between stations?

If we define the minimum distance required for triangulation, could that help participants decide whether it would be worth their time and money to build and place more than station for example on their private property?

How much does triangulation accuracy really matter for the UAP tracking problem?

Could it be OK to have relatively high triangulation error, assuming the observed motions are extreme?

Consider an object triangulated at roughly somewhere between 5km and 10km in altitude. Its motion in image space implies a velocity roughly between Mach 10 and Mach 20. Does it matter that the triangulation error could be as high as 100 percent?
